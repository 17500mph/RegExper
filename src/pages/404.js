import React from 'react';
import PropTypes from 'prop-types';
import { withTranslation, Trans } from 'react-i18next';

import Metadata from 'components/Metadata';
import Message from 'components/Message';

export const ErrorPage = ({ t }) => <>
  <Metadata title={ t('Page Not Found') } />
  <Message type="error" heading={ t('404 Page Not Found') }>
    <p><Trans>The page you have requested could not be found.</Trans></p>
  </Message>
</>;

ErrorPage.propTypes = {
  t: PropTypes.func.isRequired
};

export default withTranslation()(ErrorPage);
