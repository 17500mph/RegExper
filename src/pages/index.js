import React from 'react';
import PropTypes from 'prop-types';
import { Link, graphql } from 'gatsby';
import URLSearchParams from '@ungap/url-search-params';

import Metadata from 'components/Metadata';
import Message from 'components/Message';
import App from 'components/App';

export const query = graphql`
  query IndexPageQuery {
    site {
      siteMetadata {
        defaultSyntax
        syntaxList { id, label }
      }
    }
  }
`;

const readURLHash = (location, defaultSyntax) => {
  const query = location.hash.slice(1);
  const params = new URLSearchParams(query);
  const permalinkUrl = query ? location.href : null;

  if (params.get('syntax')) {
    return {
      syntax: params.get('syntax'),
      expr: params.get('expr'),
      permalinkUrl
    };
  } else {
    // Assuming old-style URL
    return {
      syntax: defaultSyntax,
      expr: query,
      permalinkUrl
    };
  }
};

export const IndexPage = ({
  location,
  data: { site: { siteMetadata } }
}) => <>
  <Metadata/>
  <noscript>
    <Message type="error" heading="JavaScript Required">
      <p>You need JavaScript to use Regexper.</p>
      <p>If you have concerns regarding the use of tracking code on Regexper,
        please see the <Link to="/privacy">Privacy Policy</Link>.</p>
    </Message>
  </noscript>
  <App
    syntaxList={ siteMetadata.syntaxList }
    { ...readURLHash(location, siteMetadata.defaultSyntax) } />
</>;

IndexPage.propTypes = {
  location: PropTypes.object,
  data: PropTypes.shape({
    site: PropTypes.shape({
      siteMetadata: PropTypes.shape({
        defaultSyntax: PropTypes.string,
        syntaxList: PropTypes.arrayOf(PropTypes.shape({
          id: PropTypes.string,
          label: PropTypes.string
        }))
      })
    })
  })
};

export default IndexPage;
