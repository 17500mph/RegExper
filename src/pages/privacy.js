import React from 'react';
import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';

import Metadata from 'components/Metadata';
import PrivacyPolicy from 'components/PrivacyPolicy';

export const PrivacyPage = ({ t }) => <>
  <Metadata title={ t('Privacy Policy') } />
  <PrivacyPolicy />
</>;

PrivacyPage.propTypes = {
  t: PropTypes.func.isRequired
};

export default withTranslation()(PrivacyPage);
