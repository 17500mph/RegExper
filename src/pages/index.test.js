import React from 'react';
import { shallow } from 'enzyme';

import { IndexPageImpl } from 'pages/index';

describe('Index Page', () => {
  test('rendering', () => {
    const component = shallow(
      <IndexPageImpl />
    );
    expect(component).toMatchSnapshot();
  });
});
