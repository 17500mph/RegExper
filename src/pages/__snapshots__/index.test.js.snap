// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page rendering 1`] = `
<Fragment>
  <withI18nextTranslation(Metadata)
    description="Test description"
  />
  <noscript>
    <Message
      heading="JavaScript Required"
      type="error"
    >
      <p>
        You need JavaScript to use Regexper.
      </p>
      <p>
        If you have concerns regarding the use of tracking code on Regexper, please see the 
        <mockConstructor
          to="/privacy"
        >
          Privacy Policy
        </mockConstructor>
        .
      </p>
    </Message>
  </noscript>
  <withI18nextTranslation(App)
    expr=""
    permalinkUrl={null}
    syntax="testJs"
    syntaxList={
      Array [
        Object {
          "id": "testJS",
          "name": "Testing JS",
        },
        Object {
          "id": "other",
          "name": "Other",
        },
      ]
    }
  />
</Fragment>
`;

exports[`Index Page rendering with an expression on the URL 1`] = `
<Fragment>
  <withI18nextTranslation(Metadata)
    description="Test description"
  />
  <noscript>
    <Message
      heading="JavaScript Required"
      type="error"
    >
      <p>
        You need JavaScript to use Regexper.
      </p>
      <p>
        If you have concerns regarding the use of tracking code on Regexper, please see the 
        <mockConstructor
          to="/privacy"
        >
          Privacy Policy
        </mockConstructor>
        .
      </p>
    </Message>
  </noscript>
  <withI18nextTranslation(App)
    expr="testing"
    permalinkUrl="http://example.com"
    syntax="test"
    syntaxList={
      Array [
        Object {
          "id": "testJS",
          "name": "Testing JS",
        },
        Object {
          "id": "other",
          "name": "Other",
        },
      ]
    }
  />
</Fragment>
`;
